### ISSUE TITLE
bug(linter): false negative - Missing "key" prop for element in iterator.

### ISSUE THREAD

**Issue by Newbie012** [web-infra-dev/oxc#1183]
Getting:

```
⚠ eslint-plugin-react(jsx-key): Missing "key" prop for element in iterator.
    ╭─[src/shared/components/AddFieldSelection.tsx:39:1]
 39 │           title: "Basic",
 40 │           items: basicFieldTypes.map((fieldType) => {
    ·                                  ─┬─
    ·                                   ╰── Iterator starts here
 41 │             return {
 42 │               value: { type: fieldType.name },
 43 │               title: fieldType.label,
 44 │               icon: <FieldIcon field={fieldType} />,
    ·                      ────┬────
    ·                          ╰── Element generated here
 45 │             };
    ╰────
  help: Add a "key" prop to the element in the iterator (https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key).
```

on:

```ts
  return (
    <...
      options={[
        {
          title: "Basic",
          items: basicFieldTypes.map((fieldType) => {
            return {
              value: { type: fieldType.name },
              title: fieldType.label,
              icon: <FieldIcon field={fieldType} />,
            };
          }),
        },
        ...
```

**Comment by Boshen** [comment]
Same as https://github.com/web-infra-dev/oxc/issues/1178 I think, cc @camc314

**Comment by camc314** [comment]
Thanks, will take a look

**Comment by camc314** [comment]
Looks like both of these were fixed in #1139

**Comment by Boshen** [comment]
This is addressed in https://github.com/web-infra-dev/oxc/releases/tag/oxlint_v0.0.16
